src/ICSharpCode.SharpZipLib/Core/Exceptions/SharpZipBaseException.cs

```diff
@@ -36,5 +38,21 @@ public SharpZipBaseException(string message, Exception innerException)
 			: base(message, innerException)
 		{
 		}
+
+		/// <summary>
+		/// Initializes a new instance of the SharpZipBaseException class with serialized data.
+		/// </summary>
+		/// <param name="info">
+		/// The System.Runtime.Serialization.SerializationInfo that holds the serialized
+		/// object data about the exception being thrown.
+		/// </param>
+		/// <param name="context">
+		/// The System.Runtime.Serialization.StreamingContext that contains contextual information
+		/// about the source or destination.
+		/// </param>
+		protected SharpZipBaseException(SerializationInfo info, StreamingContext context)
```

Comment:
This was actually needed since we're inheriting from Exception (which implements `ISerializable`). But at least we can just chain the base constructor.